src/query/storage/m3/cluster_resolver.go

```diff
@@ -43,7 +42,15 @@ const (
 type unaggregatedNamespaceDetails struct {
 	satisfies        unaggregatedNamespaceType
 	clusterNamespace ClusterNamespace
-	retention        time.Duration
+}
+
+type resolvedNamespace struct {
+	ClusterNamespace
+	narrowing narrowing
+}
+
+func resolved(ns ClusterNamespace) resolvedNamespace {
```

Comment:
nit: maybe we should call this `newResolvedNamespace` by convention since essentially this is a constructor?

Reply:
Maybe let's leave it short and fluent as long as it's just a plain wrapper? Otherwise this
```go
[]resolvedNamespace{resolved(unaggregated.clusterNamespace)}, nil
```
would turn into
```go
[]resolvedNamespace{newResolvedNamespace(unaggregated.clusterNamespace)}, nil
```